include/boost/math/special_functions/gamma.hpp

```diff
@@ -1627,7 +1631,7 @@ BOOST_MATH_GPU_ENABLED T gamma_incomplete_imp_final(T a, T x, bool normalised, b
       #endif
       result = gam - result;
    }
-   if(p_derivative)
+   if(p_derivative && x > 0)
```

Comment:
I think we need to let this line execute without the check for `x > 0`, that way our root finder will get a derivative back: in this particular case, any arbitrary large value will do.... ah but there should be an `else` before the `*p_derivative /= x;`.

Reply:
Hi, I adjusted the coding. Is it okay now? 

I also corrected some wrong ternary operators in the file which prevent using some functions with non-standard float/double T template parameters.